### PR TITLE
Bump moby vendor to 14ce1f1cf48e9859223c6311de58aec4dc

### PR DESCRIPTION
**Pull request by thaJeztah**
This includes  a fix for API version negotiation: https://github.com/moby/moby/pull/35008

~~I may need some help on this one w.r.t. the local changes needed; unit-tests hung for me locally, so likely I messed something up, but pushed here as a start~~

edit: looks like hanging may be just with me locally; I'll have to check why..

### CODE REVIEW

File: cli/command/container/cp.go [hunk 1]
```diff
@@ -113,7 +113,7 @@ func resolveLocalPath(localPath string) (absPath string, err error) {
 		return
 	}
 
-	return archive.PreserveTrailingDotOrSeparator(absPath, localPath), nil
+	return archive.PreserveTrailingDotOrSeparator(absPath, localPath, filepath.Separator), nil
```
Comment (thaJeztah) [inline cli/command/container/cp.go]: ping @johnstep to check if this one is correct (LCOW related)

**Comment by thaJeztah** [comment]
ping @dnephin @vdemeester if we want this updated before 17.10 is cut off

**Comment by thaJeztah** [comment]
FYI: running locally (D4M), it got stuck at;

```
--- PASS: TestResolveService (0.00s)
PASS
ok  	github.com/docker/cli/cli/command/idresolver	0.027s
```

**Comment by dnephin** [comment]
> looks like hanging may be just with me locally; I'll have to check why..

@thaJeztah `TestRunBuildFromGitHubSpecialCase` added in #390 prompts for input if run with a tty :angel: This prompt is sometimes hidden by other test output , which makes it look like it's hanging.

File: vendor.conf [hunk 2]
```diff
@@ -4,17 +4,25 @@ github.com/coreos/etcd 824277cb3a577a0e8c829ca9ec557b973fe06d20
...
 github.com/docker/docker-credential-helpers 3c90bd29a46b943b2a9842987b58fb91a7c1819b
 
+# docker/docker and dependencies
+github.com/docker/docker 14ce1f1cf48e9859223c6311de58aec4dc0f046c
+github.com/containerd/continuity 22694c680ee48fb8f50015b44618517e2bde77e8
+
+# TODO remove this once tonistiigi/fsutil switched, and was vendored (https://github.com/tonistiigi/fsutil/pull/15)
+github.com/stevvooe/continuity 22694c680ee48fb8f50015b44618517e2bde77e8
+github.com/moby/buildkit aaff9d591ef128560018433fe61beb802e149de8
+github.com/tonistiigi/fsutil 1dedf6e90084bd88c4c518a15e68a37ed1370203
```
Comment (dnephin) [inline vendor.conf]: Let's keep this file sorted alphabetically instead of these adhoc groupings

**Comment by thaJeztah** [comment]
> prompts for input if run with a tty

Oh! I saw that popup, then go away so thought it worked; yes, that would explain it!

File: vendor.conf (hunk 2 above)
Comment (thaJeztah) [reply vendor.conf]: Ok, let me update 👍

**Comment by thaJeztah** [comment]
vendor.conf updated to be in alphabetical order again; PTAL

**Comment by thaJeztah** [comment]
ready to go, or do we need a check from @johnstep for https://github.com/docker/cli/pull/570#discussion_r141858117 ?

**Comment by thaJeztah** [comment]
discussed offline with @dnephin ^^

**Pull request merged**